accepted/single-file/design.md

```diff
@@ -66,225 +32,254 @@ Publishing to a single file can be triggered by adding the following property to
-
-#### VS Integration
-
-Developers should be able to use the feature easily from Visual Studio. The feature will start with text based support -- by explicitly setting the `PublishSingleFile` property in the project file. In future, we may provide a single-file publish-profile, or other UI triggers.
+* **Compression**: Currently the bundler does not compress the contents embedded at the end of the host binary.  Compressing the bundled files and meta-data can significantly reduce the size of the single-file output (by about 30%-50% as determined by prototyping). 
+* **Single-file Plugins** Extended the above design to seamlessly support single-file publish for plugins.
 
 ## User Experience
 
-To summarize, here's the overall experience for creating a HelloWorld single-file app 
-
-*  Create a new HelloWorld app: `HelloWorld$ dotnet new console`
-
-#### Framework Dependent HelloWorld
-
-* Normal publish: `dotnet publish` 
-
-  * Publish directory contains the host `HelloWorld.exe` ,  the app `HelloWorld.dll`, configuration files `HelloWorld.deps.json`, `HelloWorld.runtimeconfig.json`, and the PDB-file `HelloWorld.pdb`.
-
-* Single-file publish: `dotnet publish -r win10-x64 --self-contained=false /p:PublishSingleFile=true`
-
-  * Publish directory contains: `HelloWorld.exe` `HelloWorld.pdb`
-
-  * `HelloWorld.dll`, `HelloWorld.deps.json`, and `HelloWorld.runtimeconfig.json` are embedded within `HelloWorld.exe`.
-
-* Run: `HelloWorld.exe`
-
-  * The app runs completely from the single-file, without the need for intermediate extraction to file.
-
-#### Self-Contained HelloWorld
-
-- Normal publish: `dotnet publish -r win10-x64`
-
-  * Publish directory contains 221 files including the host, the app, configuration files, the PDB-file and the runtime.
-
-- Single-file publish: `dotnet publish -r win10-x64 /p:PublishSingleFile=true`
-
-  - Publish directory contains: `HelloWorld.exe` `HelloWorld.pdb`
-  - The remaining 219 files are embedded within the host `HelloWorld.exe`.
-
-- Run: `HelloWorld.exe`
+To summarize, here's the overall experience for creating a HelloWorld single-file app:
 
-  * The bundled app and configuration files are processed directly from the bundle.
-  * Remaining 216 files will be extracted to disk at startup. 
-  * If reuse of extracted files is enabled, subsequent runs of the app may skip the extraction step.
+* Create a new HelloWorld app: `HelloWorld$ dotnet new console`
 
+* Framework Dependent Publish
 
-Most applications are expected to work without any changes. However, apps with a strong expectation about absolute location of dependent files may need to be made aware of bundling and extraction aspects of single-file publishing. No difference is expected with respect to debugging and analysis of apps.
+  * Normal publish: `dotnet publish` 
+    * Published files: `HelloWorld.exe`, `HelloWorld.dll`, `HelloWorld.deps.json`, `HelloWorld.runtimeconfig.json`, `HelloWorld.pdb`
```

Comment:
BTW: There is a proposal from @richlander to get rid of `HelloWorld.deps.json` and to embed `HelloWorld.runtimeconfig.json` by default.

Reply:
We probably can't drop `.deps.json` in all cases (Framework assembly overrides for example). And we're discussing that ideally the "embedding" would be the same technology for single-file and non-single-file.

Reply:
Also, `,deps.json` file is used for assembly resolution when loading plugins.